### PR TITLE
Feature/split contributions

### PR DESCRIPTION
**Pull request by vstam1**
## What?
- Split extrinsics into community and remainder contributions
- Simplify contribute benchmarks and weight returns
- call inner do_functions in tests

## Why?
- In the community, we need to check if the user has a winning bid already, and if so return err. The credential check should be the same though.
  - The token amount left to buy is different in each round
- Was too complex, making it harder to maintain. It's now less accurate but its a worthy tradeoff
- Extrinsic calls will do credential checks, and we don't want those in pallet tests. We do those in integration tests 

## How?
- abstract common logic into private function, create 2 public functions with a distinct logic
- assume we always create the ct account, and we always do some sort of iteration over users. We now have only 2 logic paths, either the contribution ends the round or it doesn't. The rest of the logic branches get included in the linear interpolation (even though it's not a linear relationship)

## Testing?
- Since we made the original do_contribute private, we are now forced to replace all calls with either community or remainder contribute calls. So just run `cargo test`
- Run benchmark tests and benchmark weight generation and check results.
- Add a test for the winning bidder to not be accepted in the community but in the remainder.

## Anything Else?
Next PR should make the default values used in tests to be more complex, to be in line with the new logic of who gets to participate in the community.

### CODE REVIEW

**Review by JuaniRios — APPROVED** [review]
I coded the last part of the PR so LGTM

File: integration-tests/src/tests/basic_comms.rs
Comment (lrazovic) [inline integration-tests/src/tests/basic_comms.rs:1]: What's the reasoning behind these changes?

File: integration-tests/src/tests/mod.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+// infinite loop
```
Comment (lrazovic) [inline integration-tests/src/tests/mod.rs:17]:
```suggestion
```

File: pallets/funding/src/weights_old.rs
Comment (lrazovic) [inline pallets/funding/src/weights_old.rs:1]: Can we remove this file?

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "1.74.0"
+channel = "nightly-2023-12-01"
```
Comment (lrazovic) [inline rust-toolchain.toml:2]:
```suggestion
channel = "1.74.0"
```

**Review by lrazovic — CHANGES_REQUESTED** [review]

File: integration-tests/src/tests/basic_comms.rs
Comment (JuaniRios) [reply integration-tests/src/tests/basic_comms.rs:1]: oops, forgot to uncomment

File: pallets/funding/src/weights_old.rs
Comment (JuaniRios) [reply pallets/funding/src/weights_old.rs:1]: yes

File: integration-tests/src/tests/basic_comms.rs [hunk 3]
```diff
@@ -20,6 +20,9 @@ use crate::*;
 const MAX_REF_TIME: u64 = 300_000_000;
 const MAX_PROOF_SIZE: u64 = 10_000;
 
+#[test]
+fn test() {}
+
```
Comment (lrazovic) [inline integration-tests/src/tests/basic_comms.rs:25]: Is this needed?

**Review by lrazovic — APPROVED** [review]
One last very minor comment, but great simplification!

File: integration-tests/src/tests/basic_comms.rs (hunk 3 above)
Comment (JuaniRios) [reply integration-tests/src/tests/basic_comms.rs]: good catch!

**Pull request merged**